src/ORM/Table.php

```diff
@@ -2101,7 +2101,6 @@ protected function _saveMany(iterable $entities, $options = []): iterable
         /** @var bool[] $isNew */
         $isNew = [];
         $cleanup = function ($entities) use (&$isNew): void {
-            /** @var \Cake\Datasource\EntityInterface[] $entities */
```

Comment:
Sure those are not needed for IDE?

Reply:
Neither phpstan or psalm needs this annotation. In general we should only worry about type inference by IDEs for  methods/properties used by devs, not framework internals.

Reply:
Those contribute to understanding and ide refactoring.
I strongly disagree with this sentence.

Reply:
If you want to remove potentielly useful and needed annotations please never do so without using an IDE to back it up and to avoid regressions where not necessary.
A change to remove sth working for the sake of it is a bad change.

Reply:
Working = followable clickable method link etc.

Reply:
Do different IDEs handle type inference differently? Could testing in one still result in another breaking?

Reply:
IDE auto completions are a convenience not a necessity. So while they might be good to have they aren't essentially *needed*.

> Do different IDEs handle type inference differently?

Most have their own static analyzers so they would have varying levels of type inference, though most people who heavily reply on IDE auto completion features are phpstorm users.

> Could testing in one still result in another breaking?

Does any PHP IDE provide a way to use its static analysis feature from command line? If something isn't testable through CI then supporting it is not feasible.

Reply:
![issue2](https://user-images.githubusercontent.com/39854/66736867-b610a900-ee6a-11e9-83f2-10db61059afc.png)

>  If something isn't testable through CI then supporting it is not feasible.

That is so wrong. That statement in itself explains why this PR and the intension is bad by design.
We do not code nice and shiny documented code for machines or CI cystems, but for humans. You seem to have forgotten that. Just because CI tooling does seem to not care about this, you take away something now from IDE tooling and human usability that people have put work and effort and time into to have.
We worked all hard to get IDE compatibility to a reasonable level and PRs like this just compromise this. It is a slap in the face to all those that put in that effort to get a nice core code to click and read through.
:disappointed:

Reply:
He didn't test it. He just blindly removed because CI system didn't complain.
If at least one IDE needs them it is worth keeping for that one, and in no way ever removing a valid one just for the sake of removing.